test/integration/targets/nxos_snapshot/tests/common/sanity.yaml

```diff
@@ -0,0 +1,39 @@
+---
+- debug: msg="START TRANSPORT:{{ connection.transport }} nxos_snapshot sanity test"
+
+- block:
+  - name: create snapshot
+    nxos_snapshot:
+      action: create
+      snapshot_name: test_snapshot1
+      description: Ansible
+      provider: "{{ connection }}"
+
+  - name: create another snapshot
+    nxos_snapshot:
+      action: create
+      snapshot_name: test_snapshot2
+      description: row
+      section: myshow
+      show_command: show ip interface brief
+      row_id: ROW_intf
+      element_key1: intf-name
+      provider: "{{ connection }}"
+
+  - name: compare snapshots
+    nxos_snapshot:
+      action: compare
+      snapshot1: test_snapshot1
+      snapshot2: test_snapshot2
+      comparison_results_file: compare_snapshots.txt
+      compare_option: summary
+      path: '.'
+      provider: "{{ connection }}"
+
+  - name: delete snapshot
+    nxos_snapshot:
+      action: delete_all
+      provider: "{{ connection }}"
+  when: (titanium is defined) and not ((titanium | search('true')))
```

Comment:
This doesn't have to be skipped on both transports, cli is fine. a check in nxapi/sanity.yaml would be more precise.

Reply:
just hold the PR, until I retest this and clean it for nxapi only. Having difficulty in getting the proper when condition.